classes/page_group.php

```diff
@@ -54,6 +89,36 @@ protected function set_fuzzydurationcounts(array $data) {
         $this->raw_set('fuzzydurationcounts', json_encode($data, JSON_FORCE_OBJECT));
     }
 
+    /**
+     * Called after the page group is created in the DB.
+     */
+    public function after_create() {
+        $this->update_cache();
+    }
+
+    /**
+     * Called after the DB is updated.
+     *
+     * @param bool $result Whether or not the update was successful.
+     */
+    public function after_update($result) {
+        if ($result) {
+            $this->update_cache();
+        }
+    }
+
+    /**
+     * Updates the cache with the current record.
+     *
+     * @throws \coding_exception
+     */
+    protected function update_cache() {
+        $keydata = ['name' => $this->get('name'), 'month' => $this->get('month')];
```

Comment:
Small thing - which is a personal opinion. It may be slightly clearer to see the key values of this multi-value array if they were put on multiple lines. The reasoning behind the suggestion is because the values aren't simple variables, and instead are calls to methods with params.

For example:
```suggestion
        $keydata = [
            'name' => $this->get('name'), 
            'month' => $this->get('month'),
        ];
```